src/QueryResults.ts

```diff
@@ -0,0 +1,62 @@
+import Promise, { Thenable, Executor } from 'dojo-core/Promise';
+import Task from 'dojo-core/async/Task';
+import {duplicate, mixin} from 'dojo-core/lang';
+import * as dstore from './interfaces';
+
+export interface QueryOptions {
+	totalLength?: number| Promise<number>;
+	response: any;
```

Comment:
You're correct, it shouldn't be required. What's the correct way to spell out each permutation?